pkg/pool-weighted/test/foundry/LiquidityApproximationWeighted.t.sol

```diff
@@ -88,413 +90,184 @@ contract LiquidityApproximationWeightedTest is LiquidityApproximationTest, Weigh
         swapFeePercentage = bound(swapFeePercentage, minSwapFeePercentage, maxSwapFeePercentage);
-        weightDai = bound(weightDai, 1e16, 99e16);
 
-        // Weights can introduce some differences in the swap fees calculated by the pool during unbalanced add/remove
-        // liquidity, so the error tolerance needs to be a bit higher than the default tolerance.
-        defectRoundingDelta = 0.0001e16; // 0.0001%
+        setSwapFeePercentageInPools(swapFeePercentage);
 
         // For very small invariant ratios, `BasePoolMath` reverts when calculated amount out < 0 because of rounding.
         // Perform an external call so that `expectRevert` catches the error.
         vm.expectRevert(stdError.arithmeticError);
-        this.removeLiquiditySingleTokenExactInWeights(exactBptAmountIn, swapFeePercentage, weightDai);
-    }
-
-    function removeLiquiditySingleTokenExactInWeights(
-        uint256 exactBptAmountIn,
-        uint256 swapFeePercentage,
-        uint256 weightDai
-    ) public {
-        _setPoolBalancesWithDifferentWeights(weightDai);
-
-        uint256 amountOut = removeExactInArbitraryBptIn(exactBptAmountIn, swapFeePercentage);
-        swapFeePercentage > 0
-            ? assertLiquidityOperation(amountOut, swapFeePercentage, false)
-            : assertLiquidityOperationNoSwapFee();
+        this.removeExactInArbitraryBptInExternal(exactBptAmountIn);
     }
 
     // Utils
-
-    function _computeMaxTokenAmount(uint256 weightDai) private view returns (uint256 maxAmount) {
-        // `maxAmount` must be lower than 30% of the lowest pool liquidity. Below, `maxAmount` is calculated as 25%
-        // of the lowest liquidity to have some error margin.
-        maxAmount = weightDai > 50e16
-            ? poolInitAmount.mulDown(weightDai.complement())
-            : poolInitAmount.mulDown(weightDai);
-        maxAmount = maxAmount.mulDown(25e16);
-    }
-
-    function _computeMaxBptAmount(
-        uint256 weightDai,
-        uint256 swapFeePercentage
-    ) private view returns (uint256 maxAmount) {
-        uint256 totalSupply = IERC20(liquidityPool).totalSupply();
-        // Compute the portion of the BPT supply that corresponds to the DAI tokens.
-        uint256 daiSupply = totalSupply.mulDown(weightDai);
-        // When we add liquidity unbalanced, fees will make the Vault request more tokens.
-        // We need to offset this effect: we want to bring down the max amount even further when fees are larger,
-        // so we multiply the DAI supply with a lower value as fees go higher.
-        uint256 daiSupplyAccountingFees = daiSupply.mulDown(swapFeePercentage.complement());
-
-        // Finally we multiply by 25% (30% is max in ratio, this leaves some margin for error).
-        maxAmount = daiSupplyAccountingFees.mulDown(25e16);
+    /**
+     * @dev vm.expectRevert only captures the error of the next external call. If we call an internal function with
```

Comment:
Do we need to repeat this comment for every test? Maybe make a block comment above and say it applies to all the `...External` functions?

Reply:
Done!